### PR TITLE
Update to new 2019-01-19 nightly.

### PR DESCRIPTION
**Pull request by shaleh**
Update external crate versions where possible.
Uses official bindgen 0.46 instead of git repo hash.
Use full path in call to `defsubr` in the macro so it does not need to be imported everywhere.
Fix clippy and rustfmt complaints.
Remove some dead code.

### CODE REVIEW

File: rust_src/src/crypto/mod.rs [hunk 1]
```diff
@@ -63,205 +52,6 @@ fn hash_alg(algorithm: LispSymbolRef) -> HashAlg {
     }
 }
```
Comment (shaleh) [inline rust_src/src/crypto/mod.rs]: This was all waiting for `extract_data_from_object` to be ported. No sense keeping this zombie alive.

**Comment by shaleh** [comment]
This is a noisy patch with little actual change.

**Comment by daichimukai** [comment]
Well, nightly-2019-01-21 and nightly-2019-01-22 lack clippy component at least for x86_64 linux system?

[Reference](https://mexus.github.io/rustup-components-history/x86_64-unknown-linux-gnu.html)

**Comment by brotzeit** [comment]
@daichimukai Interesting.

**Comment by agraven** [comment]
@brozeit it's pretty inconsistent which nightly versions have what tools. There's a page listing the latest nightly each component supports, but I can't remember exactly where to find it

**Comment by shaleh** [comment]
brotzeit's rebase did not appear to be happy. I ran it again from my original branch. Let's see how this one goes.

**Comment by shaleh** [comment]
Stuck on cerrors timing out....

**Comment by shaleh** [comment]
Ok, so something is up with the `cerror` checks apparently. Looks like it is time to revisit.

I have what I hope is a quick solution in mind. We can add a new Makefile target in `src/` which only does the C build but none of the Rust. The core build will depend on this target.

**Comment by shaleh** [comment]
Nevermind that about the Makefile. I misinterpreted the 'cerrors' name. The actual intent is to verify that `lisp_fn` macros show proper details. Now to see if we can simplify this.

**Comment by shaleh** [comment]
This will be rebased on #1280 when it is approved.

**Pull request merged**